File: bulldog-annotations/src/main/java/com/criations/bulldog_annotations/Bulldog.java
package com.criations.bulldog_annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks an interface as a target for Bulldog annotation processor
 * <pre><code>
 * {@literal @}
 * @Bulldog
 * interface Settings{
 *      // values
 *  }
 * </code></pre>
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface Bulldog {
}

File: bulldog-annotations/src/main/java/com/criations/bulldog_annotations/Bulldog.kt
package com.criations.bulldog_annotations

/**
 * Marks an interface as a target for Bulldog annotation processor.
 * <pre>`
 * @Bulldog
 * interface Settings {
 * val id: Int
 * val email: String
 * }
 * `</pre>
 */
@Target(AnnotationTarget.CLASS)
@Retention(value = AnnotationRetention.RUNTIME)
annotation class Bulldog(val name: String = "")

@Target(AnnotationTarget.FIELD)
@Retention(value = AnnotationRetention.RUNTIME)
annotation class Enum(val value: String)

